### PR TITLE
test: resolve CI failures

### PR DESCRIPTION
**Pull request by David-Wobrock**
## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ ] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got the approval (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](https://github.com/ory/docs).

### CODE REVIEW

File: selfservice/strategy/code/strategy_login.go [hunk 1]
```diff
@@ -303,13 +290,11 @@ func (s *Strategy) findIdentityForIdentifier(ctx context.Context, identifier str
...
 				}
 
-				address, err := s.findIdentifierInVerifiableAddress(session.Identity, identifier)
+				_, err := s.findIdentifierInVerifiableAddress(session.Identity, identifier)
 				if err != nil {
 					return nil, nil, err
 				}
 
-				addresses = []Address{*address}
-
```
Comment (aeneasr) [inline selfservice/strategy/code/strategy_login.go:312]: Revert this please

Comment (David-Wobrock) [reply selfservice/strategy/code/strategy_login.go]: Funny how the linter doesn't pick this up properly 👀

Comment (aeneasr) [reply selfservice/strategy/code/strategy_login.go]: My bad, it is indeed in-effective.

**Review by aeneasr — APPROVED** [review]

File: test/e2e/cypress/integration/profiles/code/login/error.spec.ts [hunk 2]
```diff
@@ -125,7 +125,7 @@ context("Login error messages with code method", () => {
         )
       })
 
-      it("should show error message when required fields are missing", () => {
+      it.only("should show error message when required fields are missing", () => {
```
Comment (Saancreed) [inline test/e2e/cypress/integration/profiles/code/login/error.spec.ts:128]: Won't that `only` here prevent other tests from being executed? Or is it actually the intent here?

Comment (jonas-jonas) [reply test/e2e/cypress/integration/profiles/code/login/error.spec.ts:128]: Good point, cypress unfortunately has no tooling around preventing this in CI. :/

```suggestion
      it("should show error message when required fields are missing", () => {
```

**Review by jonas-jonas — CHANGES_REQUESTED** [review]

**Review by jonas-jonas — APPROVED** [review]

File: identity/credentials.go [hunk 3]
```diff
@@ -248,7 +249,13 @@ func CredentialsEqual(a, b map[CredentialsType]Credentials) bool {
 			return false
 		}
 
-		if string(expect.Config) != string(actual.Config) {
+		// Try to normalize configs (remove spaces etc).
+		patch, err := jsondiff.CompareJSON(actual.Config, expect.Config)
```
Comment (aeneasr) [inline identity/credentials.go:253]: This is a comprehensive comparison, ensuring that key shuffling or other things (e.g. extra spaces) don't cause trouble!

File: test/e2e/cypress/integration/profiles/code/login/error.spec.ts [hunk 4]
```diff
@@ -160,6 +160,12 @@ context("Login error messages with code method", () => {
             "contain",
             "Property identifier is missing",
           )
+        } else if (app === "react") {
+          // The backspace trick is not working in React.
+          cy.get('[data-testid="ui/message/4010008"]').should(
```
Comment (aeneasr) [inline test/e2e/cypress/integration/profiles/code/login/error.spec.ts:165]: The React UI does not really allow deleting input fields, but we're also fine if this flow fails with another failure reason (invalid code).

**Review by zepatrik — APPROVED** [review]

**Pull request merged**